src/functionHandlers/getProfile/getProfile.ts

```diff
@@ -18,9 +18,22 @@ const getIdentifier = async (event: APIGatewayEvent) => {
   if (!id) {
     throw new createHttpError.BadRequest("Identifier is not provided");
   }
+
+  const apiKey =
+    event.headers["x-api-key"] ||
+    event.headers.Authorization ||
+    event.headers["api-key"] ||
+    event.headers.apikey ||
+    config.sheetsApiKey;
+
+  if (!apiKey) {
+    throw new Error("API key is not provided");
+  }
```

Comment:
Do a simple validation of the key, not just checking that it is present. For a start you can have it in the code.